internal/services/recommendation_poller.go

```diff
@@ -0,0 +1,176 @@
+	log := logging.GetLogger()
+	db := database.GetDB()
+	tx := db.Begin()
+	defer func() {
+		if r := recover(); r != nil {
+			tx.Rollback()
+		}
+	}()
+
+	if err := tx.Error; err != nil {
+		return err
+	}
+
+	for _, recommendationSet := range recommendationSetList {
+		if err := recommendationSet.CreateRecommendationSet(tx); err != nil {
+			log.Errorf("unable to save a record into recommendation set: %v. Error: %v", recommendationSet, err)
+			tx.Rollback()
+			return err
+		} else {
+			log.Infof("%s - Recommendation saved for experiment - %s and end_interval - %s", recommendationType, experiment_name, recommendationSet.MonitoringEndTimeStr)
+		}
+	}
+	for _, historicalRecommendationSet := range histRecommendationSetList {
+		if err := historicalRecommendationSet.CreateHistoricalRecommendationSet(tx); err != nil {
+			log.Errorf("unable to get or add record to historical recommendation set table: %v. Error: %v", historicalRecommendationSet, err)
+			tx.Rollback()
+			return err
+		}
+	}
+	return tx.Commit().Error
+}
+
+func requestAndSaveRecommendation(kafkaMsg types.RecommendationKafkaMsg, recommendationType string) bool {
+	log := logging.GetLogger()
+	experiment_name := kafkaMsg.Metadata.Experiment_name
+	maxEndTimeFromReport := kafkaMsg.Metadata.Max_endtime_report
+	poll_cycle_complete := false
+
+	recommendation, err := kruize.Update_recommendations(experiment_name, maxEndTimeFromReport)
+	if err != nil {
+		end_interval := utils.ConvertDateToISO8601(maxEndTimeFromReport.String())
+		if err.Error() == fmt.Sprintf("Recommendation for timestamp - \" %s \" does not exist", end_interval) {
+			log.Infof("recommendation does not exist for timestamp - \" %s \"", end_interval)
+		}
+		log.Errorf("unable to list recommendation for: %v", err)
+		return poll_cycle_complete
+	}
+
+	// TODO: Is_valid_recommendation to be called on every container record v20.1 upgrade on wards
+	if kruize.Is_valid_recommendation(recommendation) {
+		containers := recommendation[0].Kubernetes_objects[0].Containers
+		recommendationSetList := []model.RecommendationSet{}
+		histRecommendationSetList := []model.HistoricalRecommendationSet{}
+
+		for _, container := range containers {
+			for _, v := range container.Recommendations.Data {
+				marshalData, err := json.Marshal(v)
+				if err != nil {
+					log.Errorf("unable to list recommendation for: %v", err)
+				}
+				// Create RecommendationSet entry into the table.
+				recommendationSet := model.RecommendationSet{
+					WorkloadID:          kafkaMsg.Metadata.Workload_id,
+					ContainerName:       container.Container_name,
+					MonitoringStartTime: v.Duration_based.Short_term.Monitoring_start_time,
+					MonitoringEndTime:   v.Duration_based.Short_term.Monitoring_end_time,
+					Recommendations:     marshalData,
+				}
+				recommendationSetList = append(recommendationSetList, recommendationSet)
```

Comment:
Added change!
Per container there will be one recomm. so best to end transaction/poll cycle after each recomm. per container has been saved.